qiskit_nature/transformers/second_quantization/electronic/seniority_zero_transformer.py

```diff
@@ -0,0 +1,242 @@
+
+        Args:
+            grouped_property: the `GroupedElectronicProperty` to be transformed.
+
+        Returns:
+            A new `GroupedElectronicProperty` instance.
+
+        Raises:
+            QiskitNatureError: If the provided `GroupedElectronicProperty` does not contain a
+                               `ParticleNumber` instance.
+        """
+        if not isinstance(grouped_property, GroupedElectronicProperty):
+            raise QiskitNatureError(
+                "Only `GroupedElectronicProperty` objects can be transformed by this Transformer, "
+                f"not objects of type, {type(grouped_property)}."
+            )
+
+        particle_number = grouped_property.get_property(ParticleNumber)
+        if particle_number is None:
+            raise QiskitNatureError(
+                "The provided `GroupedElectronicProperty` does not contain a `ParticleNumber` "
+                "property, which is required by this transformer!"
+            )
+        particle_number = cast(ParticleNumber, particle_number)
+
+        self._num_modes = particle_number.num_spin_orbitals // 2
+
+        grouped_property_transformed = self._transform_property(grouped_property)
+
+        return grouped_property_transformed  # type: ignore[return-value]
+
+    def _transform_property(self, prop: Property) -> Property:
+        """Transforms a `Property` object.
+
+        This is a recursive reduction, iterating `GroupedProperty` objects when encountering one.
+
+        Args:
+            property: the property object to transform.
+
+        Returns:
+            The transformed property object.
+
+        Raises:
+            TypeError: if an unexpected Property subtype is encountered.
+        """
+        transformed_property: Property
+        # Code for recursion is copied from ActivateSpaceTransformer
+        if isinstance(prop, GroupedProperty):
+            transformed_property = deepcopy(prop)
+
+            for internal_property in iter(prop):
+                try:
+                    transformed_internal_property = self._transform_property(internal_property)
+                    if transformed_internal_property is not None:
+                        transformed_property.add_property(transformed_internal_property)
+                    else:
+                        transformed_property.remove_property(internal_property)
+                except TypeError:
+                    logger.warning(
+                        "The Property %s of type %s could not be transformed!",
+                        internal_property.name,
+                        type(internal_property),
+                    )
+                    continue
+
+            if len(transformed_property._properties) == 0:
+                transformed_property = None
+
+        elif isinstance(prop, ElectronicEnergy):
+            h_1 = prop.get_electronic_integral(ElectronicBasis.MO, 1).get_matrix()
```

Comment:
I think it would be good if we raise or log a warning if the alpha and beta integrals are not identical. I.e. `if not np.allclose(....get_matrix(0), ....get_matrix(1))`